lib/puppet/trusted_external.rb

```diff
@@ -0,0 +1,9 @@
+require 'puppet/indirector'
+
+# A class for managing external trusted facts
+class Puppet::TrustedExternal
+  # Set up indirection, so that external trusted facts can be loaded
+  extend Puppet::Indirector
+
+  indirects :trusted_external, :terminus_setting => :trusted_external_terminus, :doc => "Where to find node external trusted facts."
```

Comment:
I'd prefer to not add new indirections. Can we just add a `trusted_command` setting of type `:file`, which defaults to `nil`, similar to the `autosign_command`?

Reply:
This would shut the door on (easily) adding a REST indirector further down the line. @mrzarquon would that reduction in flexibility still work for your use-case?

Reply:
I'm ok with focusing just on getting the exec working and using future httpclient feature if we discover we need a faster way to pull data from a rest interface.